nautobot/ipam/views.py

```diff
@@ -815,36 +833,55 @@ def dispatch(self, request, *args, **kwargs):
         if "interface" not in request.GET and "vminterface" not in request.GET:
             return redirect("ipam:ipaddress_add")
 
+        _, error_msg = retrieve_interface_or_vminterface_from_request(request)
+        if error_msg:
+            messages.warning(request, error_msg)
+            return redirect(request.GET.get("return_url", "ipam:ipaddress_add"))
+
         return super().dispatch(request, *args, **kwargs)
 
     def get(self, request, *args, **kwargs):
-        form = forms.IPAddressAssignForm()
+        interface, _ = retrieve_interface_or_vminterface_from_request(request)
+        form = forms.IPAddressAssignForm(data=request.GET)
+
+        table = None
+        if request.GET.get("q"):
+            addresses = self.queryset.select_related("tenant").exclude(pk__in=interface.ip_addresses.values_list("pk"))
+            table = tables.IPAddressAssignTable(addresses)
+            paginate = {
+                "paginator_class": EnhancedPaginator,
+                "per_page": get_paginate_count(request),
+            }
+            RequestConfig(request, paginate).configure(table)
+            max_page_size = get_settings_or_config("MAX_PAGE_SIZE")
+            if max_page_size and paginate["per_page"] > max_page_size:
+                messages.warning(
+                    request,
+                    f'Requested "per_page" is too large. No more than {max_page_size} items may be displayed at a time.',
+                )
```

Comment:
Why are we paginating if we render the table with `disable_pagination=True`?

Reply:
We had to do `{% include 'utilities/obj_table.html' with table_template='panel_table.html' disable_pagination=True %}` and then render the paginator ourselves later in the file. Otherwise we ended up with nested `<form>` elements which caused the paginator to submit the wrong form.